flink-runtime/src/main/java/org/apache/flink/runtime/state/ttl/TtlAggregatingState.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.state.ttl;
+
+import org.apache.flink.api.common.typeutils.TypeSerializer;
+import org.apache.flink.runtime.state.internal.InternalAggregatingState;
+
+import java.util.Collection;
+
+/**
+ * This class wraps aggregating state with TTL logic.
+ *
+ * @param <K> The type of key the state is associated to
+ * @param <N> The type of the namespace
+ * @param <IN> Type of the value added to the state
+ * @param <ACC> The type of the accumulator (intermediate aggregate state).
+ * @param <OUT> Type of the value extracted from the state
+ *
+ */
+class TtlAggregatingState<K, N, IN, ACC, OUT>
+	extends AbstractTtlState<K, N, ACC, TtlValue<ACC>, InternalAggregatingState<K, N, IN, TtlValue<ACC>, OUT>>
+	implements InternalAggregatingState<K, N, IN, ACC, OUT> {
+
+	TtlAggregatingState(
+		InternalAggregatingState<K, N, IN, TtlValue<ACC>, OUT> originalState,
+		TtlConfig config,
+		TtlTimeProvider timeProvider,
+		TypeSerializer<ACC> valueSerializer,
+		TtlAggregateFunction<IN, ACC, OUT> aggregateFunction) {
+		super(originalState, config, timeProvider, valueSerializer);
+		aggregateFunction.stateClear = originalState::clear;
```

Comment:
Maybe it is better to pass a  `TtlAggregateFunctionBuilder` and then supply `stateClear` and `updater` before the object is created. I think then they can also become immutable. Similar changes in the other states.

Reply:
Problem here is that original state factory requires a wrapped descriptor which is based on already constructed wrapped function, basically before [original state is constructed](https://github.com/apache/flink/pull/6196/commits/99994dedb9a20244a2addd337617778b17fe8349#diff-13011fbe7c28b56b994783572b461aaeR164).